### PR TITLE
Simplify image processing by removing image enhancers

### PR DESCRIPTION
**Pull request by Bond-009**

### CODE REVIEW

File: Jellyfin.Drawing.Skia/SkiaEncoder.cs [hunk 1]
```diff
@@ -235,9 +230,12 @@ private static bool HasDiacritics(string text)
...
         private bool RequiresSpecialCharacterHack(string path)
         {
-            if (_localizationManager.HasUnicodeCategory(path, UnicodeCategory.OtherLetter))
+            for (int i = 0; i < path.Length; i++)
             {
-                return true;
+                if (char.GetUnicodeCategory(path[i]) == UnicodeCategory.OtherLetter)
+                {
+                    return true;
+                }
```
Comment (JustAMan) [inline Jellyfin.Drawing.Skia/SkiaEncoder.cs]: why?

File: MediaBrowser.Controller/Drawing/IImageProcessor.cs [hunk 2]
```diff
@@ -20,20 +19,12 @@ public interface IImageProcessor
...
-        /// <value>The image enhancers.</value>
-        IReadOnlyCollection<IImageEnhancer> ImageEnhancers { get; set; }
-
         /// <summary>
         /// Gets a value indicating whether [supports image collage creation].
         /// </summary>
         /// <value><c>true</c> if [supports image collage creation]; otherwise, <c>false</c>.</value>
         bool SupportsImageCollageCreation { get; }
 
-        IImageEncoder ImageEncoder { get; set; }
```
Comment (JustAMan) [inline MediaBrowser.Controller/Drawing/IImageProcessor.cs]: why remove this method?

File: MediaBrowser.Controller/Drawing/ImageHelper.cs [hunk 3]
```diff
@@ -19,8 +19,6 @@ public static ImageDimensions GetNewImageSize(ImageProcessingOptions options, Im
             return GetSizeEstimate(options);
         }
 
-        public static IImageProcessor ImageProcessor { get; set; }
```
Comment (JustAMan) [inline MediaBrowser.Controller/Drawing/ImageHelper.cs]: why remove?

**Comment by Bond-009** [comment]
In short: that first change made the removal of those 2 fields possible and simplified the way the image encoder gets created and passed to the image processor

**Comment by JustAMan** [comment]
I get first change now, but why remove ImageProcessor and ImageEncoder fields?..

**Comment by Bond-009** [comment]
Both aren’t needed anymore. The image processor still has a field with the image encoder.

**Pull request merged**